Library/Homebrew/cmd/update.sh

```diff
@@ -0,0 +1,259 @@
+#!/bin/bash
+
+if [ -z "$HOMEBREW_BREW_FILE" ]
+then
+  echo "Error: $(basename "$0") must be called from brew!" >&2
+  exit 1
+fi
+
+if [ "$(id -u)" = "0" ] && [ "$(stat -f%u "$HOMEBREW_BREW_FILE")" != "0" ]
+then
+  cat >&2 <<EOS
+Cowardly refusing to `sudo brew update`
+You can use brew with sudo, but only if the brew executable is owned by root.
+However, this is both not recommended and completely unsupported so do so at
+your own risk.
+EOS
+  exit 1
+fi
+
+brew() {
+  "$HOMEBREW_BREW_FILE" "$@"
+}
+
+which_git() {
+  local which_git
+  which_git="$(which git 2>/dev/null)"
+  if [ -n "$which_git" ] && [ "/usr/bin/git" = "$which_git" ]
+    then
+    local active_developer_dir
+    active_developer_dir="$(/usr/bin/xcode-\select -print-path 2>/dev/null)"
+    if [ -n "$active_developer_dir" ] && [ -x "$active_developer_dir/usr/bin/git" ]
+    then
+      which_git="$active_developer_dir/usr/bin/git"
+    fi
+  fi
+  echo $which_git
+}
```

Comment:
This function seem to handle no Xcode/CLT case incorrectly. It should return nothing on such case but instead it will return `/usr/bin/git`.